pkg/exchange/ftx/exchange.go

```diff
@@ -149,7 +149,24 @@ func (e *Exchange) QueryTrades(ctx context.Context, symbol string, options *type
 }
 
 func (e *Exchange) QueryDepositHistory(ctx context.Context, asset string, since, until time.Time) (allDeposits []types.Deposit, err error) {
-	panic("implement me")
+	if err = verifySinceUntil(since, until); err != nil {
+		return nil, err
+	}
+
+	resp, err := e.newRest().DepositHistory(ctx)
+	if err != nil {
+		return nil, err
+	}
+	for _, r := range resp.Result {
+		d, err := toGlobalDeposit(r)
+		if err != nil {
+			return nil, err
+		}
+		if d.Asset == asset && !since.After(d.Time.Time()) && !until.Before(d.Time.Time()) {
```

Comment:
filter out the deposits for the duplicated tx ID here? check the Binance QueryDepositHistory?

Reply:
By the way, the txid, address are optional because the api also returns internal transfers/airdrop histories. I only can dedupe records from chain.